File: frontend/src/administrator/category/service/CategoryService.js
import http from '@/config/axios';

export default class CategoryService {
    save(category) {
        console.log(category)
        if (category.id === null || category.id === undefined) {
            return http.post("/api/v1/categories", category).then((response) => response);
        }

        return this.#update(category);
    }
    #update(category) {
        return http.put("/api/v1/categories", category).then((response) => response);
    }
    findByFilter(filterCategory) {
        return http.get(`/api/v1/categories/filter${filterCategory.toString()}`).then((response) => response);
    }
    findById(categoryId) {
        return http.get(`/api/v1/categories/${categoryId}`).then((response) => response);
    }
}

File: backend/category/api/src/main/java/br/com/systec/fintrack/category/service/CategoryService.java
package br.com.systec.fintrack.category.service;

import br.com.systec.fintrack.category.filter.FilterCategoryVO;
import br.com.systec.fintrack.category.model.Category;
import br.com.systec.fintrack.commons.exception.BaseException;
import org.springframework.data.domain.Page;

import java.lang.module.FindException;
import java.util.List;

public interface CategoryService {
    Category save(Category category) throws BaseException;
    Category update(Category category) throws BaseException;
    Category findById(Long id) throws FindException;
    Page<Category> findByFilterAndPageable(FilterCategoryVO filterCategory) throws BaseException;
    List<Category> findAll() throws BaseException;
}
